com/bl/ttt/tictactoe.py

```diff
@@ -49,15 +49,25 @@ def get_previous_turn_player(self):
         else:
             return self.player1
 
-    def get_player_move_log(self):
-        return self.player_move_log
+    def get_player_move_log(self, player_id='both'):
+        if player_id == 'both':
+            players_move_list = []
+            x_move_log = list(self.player_move_log.get('X'))
+            lx = len(x_move_log)
+            o_move_log = list(self.player_move_log.get('O'))
+            lo = len(o_move_log)
+            for i in range(max(lx, lo)):
+                if i < lx:
+                    players_move_list.append(x_move_log[i])
+                if i < lo:
+                    players_move_list.append(o_move_log[i])
+            return players_move_list
+        else:
+            player_move_list = list(self.player_move_log.get(player_id))
+            return player_move_list
```

Comment:
Checkout itertools.zip_longest ;)

Reply:
What behaviour would you like for the use case where no moves have been logged?
Perhaps the display of the following message to the user: 'There are no moves in the log.' ?

Reply:
It is convention to return an empty collection.

Reply:
Understood.